api/server/server.go

```diff
@@ -20,12 +20,20 @@ import (
 
 type ToDDApi struct {
 	cfg config.Config
+	tdb db.DatabasePackage
 }
 
-func (tapi ToDDApi) Start(cfg config.Config) {
+func (tapi ToDDApi) Start(cfg config.Config) error {
 
 	tapi.cfg = cfg
 
+	tdb, err := db.NewToddDB(tapi.cfg)
+	if err != nil {
+		return err
```

Comment:
I should have read the follow-up comments to that tweet - I misunderstood the intention behind his tweet to be something I learned a few months back in a Hardcore Go class.

In essence what I'm driving at is that when a failure occurs, I want to be able to troubleshoot the root cause. Getting a generic "failed to write key to etcd" doesn't really help me all that much. However, this could be addressed by simply having proper logging, which I'm fine with happening in a future patch if needed (since if it's not there already, it's my fault :smile: )

So rather than advocating custom errors everywhere (which is certainly NOT what I'm trying to do) I will go back through my comments and make sure that there's at least sufficient logging around those errors, so I will be able to properly identify any issues that come up.